lib/ci/index.js

```diff
@@ -180,6 +191,47 @@ App.prototype = {
     if (this.exited) return
     this.cleanExit(this.getExitCode())
     this.exited = true
+  },
+
+  addSignalListeners: function(callback) {
+    this._boundSigInterrupt = function() {
+      this.wrapUp(new Error('Received SIGINT signal'))
+    }.bind(this)
+    process.on('SIGINT', this._boundSigInterrupt)
```

Comment:
This is the best place I've figured to put these so far. Happy to move them anywhere.